File: java/src/test/java/kata/Roman.java
package kata;

import java.util.Map;
import java.util.TreeMap;

public class Roman {

    TreeMap<Integer, String> nums = new TreeMap<>() {{
        put(1000, "M");
        put(900, "CM");
        put(500, "D");
        put(400, "CD");
        put(100, "C");
        put(90, "XC");
        put(50, "L");
        put(40, "XL");
        put(10, "X");
        put(9, "IX");
        put(5, "V");
        put(4, "IV");
        put(1, "I");
    }};

    public String toRoman(int number) {
        StringBuilder romanNumber = new StringBuilder();
        while (number >= 1) {
            Map.Entry<Integer, String> correspondence = nums.floorEntry(number);
            romanNumber.append(correspondence.getValue());
            number -= correspondence.getKey();
        }
        return romanNumber.toString();
    }
}

File: typescript/src/Roman.ts
export class Roman {
≠≠
  public toRoman(numberToConvert: number) {

    let romanDigits: [number, string][] = [
      [10, 'X'],
      [9, 'IX'],
      [5, 'V'],
      [4, 'IV'],
      [1, 'I'],
    ];

    let romanNumber = "";
    romanDigits.forEach(([arabic, romanDigit]) => {
      while (numberToConvert >= arabic) {
        romanNumber += romanDigit;
        numberToConvert -= arabic;
      }
    });
    return romanNumber;
  }
}
